File: Atividade_6/Java/ex01.java
/*
 * Entradas:
 * 1. Quantidade de passageiros
 * 2. RG (ou "nao possui")
 * 3. Passagem (ou "nao possui")
 * 4. Data de nascimento que consta no RG se necessário
 * 5. Data de nascimento que consta na passagem se necessário
 * 6. Assento se necessário
 */

import java.util.Scanner;

public class ex01 {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.println("Digite a quantidade de passageiros: ");
        int qtdPassageiros = in.nextInt();
        in.nextLine(); // Consumindo a nova linha pendente

        for (int i = 0; i < qtdPassageiros; i++) {
            System.out.println("Passageiro " + (i+1) + ", tem RG? (rg ou nao possui): ");
            String rg = in.nextLine();
            if (rg.equals("nao possui")) {
                System.out.println("A saída é nessa direção.");
                continue;       
            }

            System.out.println("Tem passagem? (passagem ou nao possui): ");
            String passagem = in.nextLine();
            if (passagem.equals("nao possui")) {
                System.out.println("A recepção é nessa direção.");
                continue;       
            }

            System.out.println("Digite a data de nascimento que consta no RG: ");
            String dataNascimentoRG = in.nextLine();
            System.out.println("Digite a data de nascimento que consta na passagem: ");
            String dataNascimentoPassagem = in.nextLine();
            if (!dataNascimentoRG.equals(dataNascimentoPassagem)) {
                System.out.println("190");
                continue;
            } 
            System.out.println("Informe seu assento: ");
            String assento = in.nextLine();
            System.out.println("O seu assento é o " + assento + ", tenha um ótimo dia.");
        }

        in.close();
    }
}

File: Atividade_6/Haskell/ex01.hs
-- Definição do tipo de dados para representar um passageiro, desnecessário nesse exercício, mas deixei para lembrar algo semelhante a struct em C
data Passenger = Passenger { rg :: String
                           , passagem :: String
                           , dataNascimentoRG :: String
                           , dataNascimentoPassagem :: String
                           , assento :: String
                           } deriving (Show)

-- Função para processar a entrada do usuário para um único passageiro
processarPassageiro :: Int -> IO ()
processarPassageiro n = do
    putStrLn $ "Passageiro " ++ show n ++ ", tem RG? (rg ou nao possui): "
    rgInput <- getLine
    if rgInput == "nao possui"
        then putStrLn "A saída é nessa direção."
        else do
            putStrLn "Tem passagem? (passagem ou nao possui): "
            passagemInput <- getLine
            if passagemInput == "nao possui"
                then putStrLn "A recepção é nessa direção."
                else do
                    putStrLn "Digite a data de nascimento que consta no RG: "
                    dataNascimentoRGInput <- getLine
                    putStrLn "Digite a data de nascimento que consta na passagem: "
                    dataNascimentoPassagemInput <- getLine
                    if dataNascimentoRGInput /= dataNascimentoPassagemInput
                        then putStrLn "190"
                        else do
                            putStrLn "Informe seu assento: "
                            assentoInput <- getLine
                            putStrLn $ "O seu assento é o " ++ assentoInput ++ ", tenha um ótimo dia."

-- Função principal para processar os passageiros
main :: IO ()
main = do
    putStrLn "Digite a quantidade de passageiros: "
    qtdPassageiros <- readLn :: IO Int
    mapM_ processarPassageiro [1..qtdPassageiros]
